app/src/main/java/com/example/pygmyhippo/organizer/EventFragment.java

```diff
@@ -0,0 +1,213 @@
+        if (event.getEventStatus().value.equals("cancelled")) {
+            closeEventButton.setBackgroundColor(0xFFA4A8C3);
+            closeEventButton.setText("Lottery closed");
+            closeEventButton.setTextColor(0xFF3A5983);
+            closeEventButton.setTextSize(20);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.organiser_fragment_event, container, false);
+        Button closeEventButton = view.findViewById(R.id.close_event_button);
+
+        closeEventButton.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view) {
+                // Only do the draw if the event hasn't already been closed
+                if (event.getEventStatus().value.equals("ongoing") && !event.getEntrants().isEmpty()) {
+                    // Change the view of the button
+                    closeEventButton.setBackgroundColor(0xFFA4A8C3);
+                    closeEventButton.setText("Lottery closed");
+                    closeEventButton.setClickable(false);
+                    closeEventButton.setTextColor(0xFF3A5983);
+                    closeEventButton.setTextSize(20);
+
+                    // Update the status of the event to closed
+                    event.setEventStatus(Event.EventStatus.cancelled);
+
+                    // Draw the lottery winners and change their statuses
+                    drawWinners(event);
+                } else if (event.getEventStatus().value.equals("ongoing") && event.getEntrants().isEmpty()) {
+                    // There are no entrants so the lottery should not be drawn
+                    Toast.makeText(getContext(), "No entrants to run the lottery on!", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        return view;
+    }
+
+    /**
+     * This method will go through the event's entrants and randomly select a specific amount to be invited
+     * @param event
+     * @author Kori
+     * TODO: Update the event in the database when entrant status gets changed
+     */
+    public void drawWinners(Event event) {
+        // Referenced Random class from https://www.geeksforgeeks.org/generating-random-numbers-in-java/
+        // Accessed on Nov. 3, 2024
+
+        Random rand = new Random();
+        int winnerNumber = 0;
+        ArrayList<Entrant> entrants = event.getEntrants();
+
+        // If the entrant list is smaller than the winner size, then just set them all as winners
+        if (entrants.size() <= event.getEventWinnersCount()) {
+            // Use the winner number as an index
+            while (winnerNumber < entrants.size()) {
+                // Update all the entrants statuses
+                //TODO: Notifications would probably be sent from here
```

Comment:
do I need to implement something related to notifications over here too, if the PR gets approved ?

Reply:
That's for the future. But yes when notifications get sent I think there is likely the spot they get sent from. Unless you see a better spot to send them. Either way it's a suggestion that in the future changing status should send notifications.